src/vs/workbench/contrib/debug/common/debugSource.ts

```diff
@@ -49,7 +49,11 @@ export class Source {
 		}
 
 		if (typeof this.raw.sourceReference === 'number' && this.raw.sourceReference > 0) {
-			this.uri = uri.parse(`${DEBUG_SCHEME}:${encodeURIComponent(path)}?session=${encodeURIComponent(sessionId)}&ref=${this.raw.sourceReference}`);
+			this.uri = uri.from({
+				scheme: DEBUG_SCHEME,
+				path,
+				query: `session=${encodeURIComponent(sessionId)}&ref=${this.raw.sourceReference}`
```

Comment:
Any special reason why you encode `sessionId`. Can this contain `=` or `&`?

Reply:
No, this should be a hex UUID. So I will remove the encode / decode part of the sessionID
Thanks

Reply:
check

Reply:
@isidorn make sure to remove the corresponding decodeURIComponent call too.

Reply:
@weinand I just removed it. Thanks